### PR TITLE
Replaced dev-dependencies with goalgoriall/open_social_dev.

### PR DESCRIPTION
**Pull request by navneet0693**
Currently, we are maintaining dev/test dependencies at different places.

1. https://github.com/goalgorilla/open_social/blob/main/composer.json#L224
2. https://github.com/goalgorilla/open_social/blob/main/tests/composer.json#L9
3. https://github.com/goalgorilla/open_social_docker/blob/master/composer.json#L30
4. https://github.com/goalgorilla/drupal_social/blob/master/composer.json#L30

This makes it difficult to update the dependencies and also to maintain them in different places. The dependencies version is the same across our repositories. 

Our inspiration comes from https://packagist.org/packages/drupal/core-dev . 

See: https://www.drupal.org/docs/develop/using-composer/starting-a-site-using-drupal-composer-project-templates#s-adding-drupalcore-dev

The new repository "https://github.com/goalgorilla/open_social_dev" contains all these dependencies required for development and testing.

### CODE REVIEW

**Comment by navneet0693** [comment]
This is currently blocked by: 

1. https://github.com/goalgorilla/open_social/pull/2749
2. https://github.com/goalgorilla/open_social/pull/2948
3. https://github.com/goalgorilla/open_social/pull/2919

File: composer.json [hunk 1]
```diff
@@ -23,28 +23,13 @@
     "require": {
         "blackfire/php-sdk": "^v1.27.1",
         "drupal/redis": "^1.5",
-        "drush/drush": "11.*@stable",
-        "goalgorilla/open_social": "dev-main",
-        "goalgorilla/open_social_scripts": "^3.0"
```
Comment (tbsiqueira) [inline composer.json:28]: why are we removing scripts?

**Review by tbsiqueira — APPROVED** [review]

File: composer.json [hunk 2]
```diff
@@ -23,28 +23,13 @@
...
+        "goalgorilla/open_social": "dev-main"
     },
     "require-dev": {
-        "dealerdirect/phpcodesniffer-composer-installer": "~0.7.1",
-        "drupal/coder": "^8.3",
-        "drupal/composer_deploy": "^1.6",
-        "drupal/core-dev": "~9.3.12",
-        "drupal/devel": "^4.1",
-        "drupal/drupal-extension": "^4.1",
         "drupal/upgrade_status": "^3.11",
```
[the remark this answers is not in the document]
Comment (navneet0693) [reply composer.json:29]: Nope few things are strongly for "development" purposes here.

File: composer.json (hunk 1 above)
Comment (navneet0693) [reply composer.json:28]: I will add scripts back.

**Review by tbsiqueira — APPROVED** [review]

**Pull request merged**